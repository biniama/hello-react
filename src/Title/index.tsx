import React, { FC, memo } from "react";

const Title: FC = () =>
    <h1>Welcome to React!!!</h1>;

/**
 * React Hooks: Memoization
 *
 * React.memo is a performance optimization tool, a higher order component. 
 * Itâ€™s similar to React.PureComponent but for function components instead of classes. 
 * If your function component renders the same result given the same props, 
 * React will memoize, skip rendering the component, and reuse the last rendered result.
 * 
 * Source: https://medium.com/@sdolidze/react-hooks-memoization-99a9a91c8853
 */
export default memo(Title);
