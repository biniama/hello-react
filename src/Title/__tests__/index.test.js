import React from "react";
import snapshotRenderer from "react-test-renderer";
import Title from "..";

describe("Title component", () => {
  it("matches Title snapshot", () => {
    const component = snapshotRenderer.create(<Title />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
