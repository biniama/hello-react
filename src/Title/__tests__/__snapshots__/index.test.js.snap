// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title component matches Title snapshot 1`] = `
<h1>
  Welcome to React!!!
</h1>
`;
